Readability.js

```diff
@@ -1082,27 +1083,47 @@ Readability.prototype = {
       if (this._debug)
         this.log("Article content after paging: " + articleContent.innerHTML);
 
+      var parseSuccessful = true;
+
       // Now that we've gone through the full algorithm, check to see if
       // we got any meaningful content. If we didn't, we may need to re-run
       // grabArticle with different flags set. This gives us a higher likelihood of
       // finding the content, and the sieve approach gives us a higher likelihood of
       // finding the -right- content.
       if (this._getInnerText(articleContent, true).length < this._wordThreshold) {
+        parseSuccessful = false;
         page.innerHTML = pageCacheHtml;
 
         if (this._flagIsActive(this.FLAG_STRIP_UNLIKELYS)) {
           this._removeFlag(this.FLAG_STRIP_UNLIKELYS);
+          this._attempts.push(articleContent);
         } else if (this._flagIsActive(this.FLAG_WEIGHT_CLASSES)) {
           this._removeFlag(this.FLAG_WEIGHT_CLASSES);
+          this._attempts.push(articleContent);
         } else if (this._flagIsActive(this.FLAG_CLEAN_CONDITIONALLY)) {
           this._removeFlag(this.FLAG_CLEAN_CONDITIONALLY);
+          this._attempts.push(articleContent);
         } else {
-          return null;
+          // No luck after removing flags, just return the longest text we found during the different loops
+          var that = this;
+          var results = this._attempts.sort(function (a, b) {
```

Comment:
Then you don't need to rebind `this` here and can just sort like so:

```js
this._attempts.sort(function(a, b) {
  return a.textLength < b.textLength;
});
```

This has the advantage that we don't recompute the text content a lot of times, and we also avoid the `this` reference.

Note that in JavaScript, `Array.sort` modifies the array on which it's called, so you don't need to assign to `var results` here.